### DISCUSSION TITLE
Sound funktioniert nicht

### DISCUSSION THREAD

**Discussion by c4sti3l** [pixelit-project/PixelIt#118] [General]
Hallo, 
ich habe wie in der Doku beschrieben den DFPlayer Mini MP3 Player angeschlossen.
Eine MP3 Datei mit dem Namen "0001.mp3" auf die SD-Karte gepackt (FAT32). 
Ich bekomme einfach keinen Ton aus dem Lautsprecher. Ich habe verschiedene SD-Karten mit 16 und 32 GB probiert, 
jedoch ohne Erfolg. Ich hoffe, ihr habt evtl. noch Tipps.

**Comment by miccgn** [comment]
Hallo,
es gibt leider mittlerweile sehr viele verschiedene Chipsätze, die für DFPlayer verwendet werden, und jeder zickt auf eine andere Weise.

Teste doch erstmal manuell, ob er überhaupt funktioniert. Auf https://alainsprojects.com/2020/03/06/df-player-mini-configuration/ findest Du ein Schaltbild. Wenn Du kurz den mit ADKEY_1 bezeichneten Pin mit GND verbindest, sollte die LED aufleuchten und der Player eine Datei abspielen. Klappt das?

**Reply by c4sti3l** [reply]
Hallo,
vielen Dank für den Tipp, ja das klappt! Jedoch immer noch nicht aus PixelIT heraus :-D

**Reply by miccgn** [reply]
Welche PixelIt-Version nutzt du denn? Und was für ein Board?

**Reply by c4sti3l** [reply]
0.3.16_beta und Wemos D1 Mini.

Ich habe mittlerweile einige Änderungen am Code und an der Pin-Belegung gemacht und die DFRobotDFPlayerMini.h statt DFPlayerMini_Fast.h verwendet, seit dem geht alles inkl. meinem BMP280.

**Reply by miccgn** [reply]
Das mit der Library überrascht mich zum einen: eigentlich sollten es beide Libraries tun.
Auf der anderen Seite bin ich wieder nicht überrascht: es gibt einige github-Seiten, die sich damit auseinandersetzen, welcher der vielen DFPlayer-Imitat-Chips welche Funktion wirklich wie unterstützt.
Offensichtlich gibt es "den Original" Player nicht mehr, aber viele Imitate, die aber mit unterschiedlicher Firmware bestückt sind. Im großen und ganzen läuft das, aber halt nicht immer. Bei dem einen helfen delay(), beim anderen das Weglassen von Widerständen, der nächste kann einfach die eine oder andere Statusabfrage gar nicht beantworten etc.

**Reply by c4sti3l** [reply]
jaa ich habe auch sehr lange rum probiert, bis ich es am laufen hatte :-D aber im Endeffekt ist wichtig, dass es nun funktioniert.

**Reply by udoht** [reply]
Hi,
ich habe mir ursrpünglich eine Awtrix2  mit DFPlayer HW247A gebaut (basierend auf dem Mainboard von Blueforcer). Vor kurzem bin ich dann auf euer Projekt umgestiegen, und bin begeistert, dass ich die alte Hardware weiter verwenden kann. 

Das einzige was ich nicht zum Laufen bekomme ist das Soundmodul. Der oben geschriebene Test mit der Groundbrücke funktioniert. Nur aus der Pixelit Oberfläche kann ich keine Sounds auslösen.

Awtrix2 benutzt D5 und D7 für die Ansteuerung des Soundmoduls. Beide Datenleitungen haben einen 1kOhm Widerstand in Reihe. Auch wenn ich die Widerstände überbrücke funktioniert der Sound nicht. 

Hat noch jemand eine Idee was ich noch ausprobieren kann?
Vielen Dank.

Gruß

Thomas

**Reply by udoht** [reply]
Ich habe gerade noch eine Beobachtung gemacht: wenn ich prev oder next track auf die Buttons lege wird das entsprechende File abgspielt. Es liegt also kein Hardware- oder Treiberproblem vor. 
Allerdings wird weder beim Booten noch via API Kommando das Abspielen ausgelöst.

**Reply by AndreasZ2** [reply]
Ich hänge mich mal hier mit an, Nutze bisher auch Awtrix2 wie udoht, mit dem gleichen Verhalten. Weder per Api, noch beim Booten wird etwas abgespielt, über die Buttons kann ich dann auch etwas abspielen. Ich habe mir zu Testzwecken von AZDelivery noch einmal einen anderen DFPlayer bestellt.

**Reply by foorschtbar** [reply]
Wie ist denn eure Verzeichnisstruktur? Für den Bootsound muss es https://pixelit-project.github.io/api.html#option-2 also \MP3\0001.mp3 sein.

**Reply by udoht** [reply]
Erst einmal danke für die Rückmeldungen. Ich habe die Verzeichnisstruktur überprüft und alles sieht gut aus.
![grafik](https://user-images.githubusercontent.com/118073558/203848535-c6753885-ec15-4d95-8446-f166292af21c.png)
Aus Verwzeiflung habe ich das "MP3" im Pfadnamen auch schon mal in "mp3" geändert, aber auch das hilft nicht.

Gruß

Thomas

**Reply by udoht** [reply]
Ich bin einen Schritt weiter. Das kein Sound abgespielt wird, liegt an der Lautstärkeeinstellung. Ein Wert <=30 funktioniert nicht. Werte größer 30 funktionieren (beide Methoden Option 1 und Option 2), haben aber keinen Einfluss auf die tatsächliche Lautstärke. Das deutet für mich auf ein Treiberproblem hin. 

Gruß

Thomas

**Reply by foorschtbar** [reply]
Durchaus, Das wurde hier schon mehrmals mit den 30% berichtet. Ich habs mal in die Doku geschrieben. Bei meinem ist selbst bei 1% volume noch was zu hören.

**Comment by udoht** [comment]
Danke für den Hinweis. Ich habe die Diskussion in dem Discord Forum gefunden. So weit ich das übersehe scheint es keine Lösung zu geben, ausser ein anderes Player Modul zu kaufen. Oder hat noch jemand gegenteilige Infos?
Danke, Gruß

Thomas

**Reply by hamster65** [reply]
Wenn es mit der funktionierenden Einstellung zu laut ist, könnte man einen Widerstand in die Leitung zum Lautsprecher setzen. Bei mir hat die Lautstärkeregelung funktioniert, aber ich musste den Play-Befehl immer ganz oft senden, bis, mit Glück, überhaupt mal eine Reaktion kam, daher habe ich es wieder ausgebaut.
